remotespark/livyclientlib/sparkcontroller.py

```diff
@@ -73,12 +74,17 @@ def add_session(self, name, connection_string, skip_if_exists, properties):
             return
 
         session = self._create_livy_session(connection_string, properties, self.ipython_display)
+
+        spark_events = SparkEvents()
```

Comment:
What about emitting the events right before we make the call to Livy and right when we get the call that says it's in running state? This includes hiveContext creation, which should be a different operation.